addons/mail/static/src/core/common/attachment_model.js

```diff
@@ -3,8 +3,9 @@
 import { assignDefined } from "@mail/utils/common/misc";
 
 import { url } from "@web/core/utils/urls";
+import { DiscussModel } from "./discuss_model";
 
-export class Attachment {
+export class Attachment extends DiscussModel {
```

Comment:
I don't like this `DiscussModel` name, it's too long and it looks like the name of a specific model related to discuss app, not a generic one

I'd suggest just `Record` as this is what those actually are.